### DISCUSSION TITLE
How Dendrite can support viewing and editing of ONNX with Transformer model architecture

### DISCUSSION THREAD

**Discussion by GeorgeS2019** [fel88/Dendrite#6] [Ideas]
Refer to these links
https://github.com/fel88/Dagre.NET/issues/2

**Comment by fel88** [comment]
@GeorgeS2019 Do you know where I can download seq2seq ONNX? Because BERT doesn't have a decoder part

I've implement some experimental logic to group nodes via prefix

Sample 1 (groups: bert/embeddings, attention, output, intermediate):
![bert_groups1](https://user-images.githubusercontent.com/15663687/135667496-0d037043-1207-47f6-a2a9-e5c9da640df8.jpg)

Sample 2 (groups: bert/embeddings, bert/encoder/layer_):
![bert_groups2](https://user-images.githubusercontent.com/15663687/135667516-19c4bb1a-8547-40ad-bfe1-a711205d9db4.jpg)

**Reply by GeorgeS2019** [reply]
![image](https://user-images.githubusercontent.com/49812372/135672684-fd6497cd-2e53-436b-821c-5ec658bc7f10.png)

I hope I mark the 12 encoder blocks. 

Suggestion:  Create a subgraph or block that organize the 12 encoder blocks (make them equal size) within one block.

One step at a time. I too am learning :-)

FYI:  use this as a [reference: **Explanation of BERT Model – NLP**](https://www.geeksforgeeks.org/explanation-of-bert-model-nlp/)
